### ISSUE TITLE
Can Some Explain the basic difference between this and big sleep? I'm a newbie

### ISSUE THREAD

**Issue by aertist** [lucidrains/deep-daze#19]

**Comment by afiaka87** [comment]
I'm new to this too! So my understanding is going to be pretty simple. 

Both projects are inspired by OpenAI's recent "DALL-E" results. They're incredibly impressive, but they haven't released the model yet and aren't likely to run on your average colab instance.

Something which has been released from that project is a tool called CLIP. CLIP can (very, very accurately) determine the contents of an image in "plain english". 

With both of these projects, we point CLIP at _another_ neural net that can generate pixels for us, and then we ask CLIP, "hey, does that look like anything in the phrase I typed in?" and it happily says "NO! That looks nothing like 'a leather armchair imitating an avocado!" The neural net feels ashamed and reflects on itself. Then it presents a new image and politely asks CLIP if it resembles this non-existent avocado chair you want to see. That cycle happens back and forth until something acceptable is produced. 

deep-daze uses the neural net SIREN as its "pixel generator". I'm not really sure how to describe SIREN in a non-mathematical way. Someone else should correct me on this, but I believe that it's a neural network designed to make complex sinusoidal waves which "solve your problem". This happens to work well for a lot of domains (including images) because of the fact that sine waves kind of describe a lot of things. As for why it specifically was chosen and works so well, I'm not sure. Hopefully someone else can chime in. 
**tl;dr** deep-daze operates on a more abstract neural net. Its results therefore tend to be less photorealistic and more "dreamlike".  

big-sleep uses BigGAN as its "pixel generator". BiGAN was built from the ground up for generating high quality imagery. It is suited specifically to that task. It has pre-existing knowledge about textures, styles, and what not. This means it is more likely to generate concrete, realistic versions of this stuff. Unfortunately, GAN's are quite brittle. They inhabit a very large knowledge space and you can very easily get incorrect results just based on it having a lousy guess.
**tl;dr** big-sleep operates with knowledge of imagery and will happily produce the textures of earth, skin, fur, etc. in order to appease CLIP.

**Comment by aertist** [comment]
Damn that's a great explanation. Thanks Dude
